File: solutions/ocaml/nth-prime/1/nth_prime.ml
let rec fn (n : int) (ps : int list) (q : int) : int =
  if List.length ps = n then List.nth ps 0 else 
  if List.for_all (fun p -> q mod p <> 0) ps then fn n (q :: ps) (q + 1) else fn n ps (q + 1)
let nth_prime n = if n <= 0 then Error "there is no zeroth prime" else Ok (fn n [] 2)

File: solutions/kotlin/nth-prime/1/src/main/kotlin/NthPrime.kt
object Prime {

    fun nth(n: Int): Int {
        if (n == 0) {
            throw IllegalArgumentException("There is no zeroth prime.")
        }
        var ps: MutableList<Int> = mutableListOf()
        var p = 2
        while (ps.size < n) {
            var flag = true
            for (q in ps) {
                if (p % q == 0) {
                    p += 1
                    flag = false
                    break
                }
            }
            if (flag) {
                ps.add(p)
                p += 1
            }
        }
        return ps[n-1]
    }
}
